marketplaceItems/appreciate/appResources/appData/appreciate_app.js

```diff
@@ -0,0 +1,949 @@
+        }
+    }
+
+    // If there exists a VR debounce timer (used for not playing sounds too often),
+    // clear it.
+    function maybeClearVRDebounceTimer() {
+        if (vrDebounceTimer) {
+            Script.clearTimeout(vrDebounceTimer);
+            vrDebounceTimer = false;
+        }
+    }
+
+    // Calculates the current intensity of appreciation based on the user's
+    // hand velocity (rotational and linear).
+    // Each type of velocity is weighted differently when determining the final intensity.
+    // The VR debounce timer length changes based on current intensity. This forces
+    // sounds to play further apart when the user isn't appreciating hard.
+    var MAX_VELOCITY_CM_PER_SEC = 110; // determined empirically
+    var MAX_ANGULAR_VELOCITY_LENGTH = 1.5; // determined empirically
+    var LINEAR_VELOCITY_WEIGHT = 0.7; // This and the line below must add up to 1.0
+    var ANGULAR_VELOCITY_LENGTH_WEIGHT = 0.3; // This and the line below must add up to 1.0
+    var vrDebounceTimer = false;
+    var VR_DEBOUNCE_TIMER_TIMEOUT_MIN_MS = 20; // determined empirically
+    var VR_DEBOUNCE_TIMER_TIMEOUT_MAX_MS = 200; // determined empirically
+    function calculateHandEffect(linearVelocity, angularVelocity){
+        var leftHandLinearVelocityCMPerSec = linearVelocity.left;
+        var rightHandLinearVelocityCMPerSec = linearVelocity.right;
+        var averageLinearVelocity = (leftHandLinearVelocityCMPerSec + rightHandLinearVelocityCMPerSec) / 2;
+        averageLinearVelocity = Math.min(averageLinearVelocity, MAX_VELOCITY_CM_PER_SEC);
+
+        var leftHandAngularVelocityLength = Vec3.length(angularVelocity.left);
+        var rightHandAngularVelocityLength = Vec3.length(angularVelocity.right);
+        var averageAngularVelocityIntensity = (leftHandAngularVelocityLength + rightHandAngularVelocityLength) / 2;
+        averageAngularVelocityIntensity = Math.min(averageAngularVelocityIntensity, MAX_ANGULAR_VELOCITY_LENGTH);
+
+        var appreciationIntensity =
+            averageLinearVelocity / MAX_VELOCITY_CM_PER_SEC * LINEAR_VELOCITY_WEIGHT +
+            averageAngularVelocityIntensity / MAX_ANGULAR_VELOCITY_LENGTH * ANGULAR_VELOCITY_LENGTH_WEIGHT;
+
+        fadeIntensity(appreciationIntensity);
+        
+        var vrDebounceTimeout = VR_DEBOUNCE_TIMER_TIMEOUT_MIN_MS +
+            (VR_DEBOUNCE_TIMER_TIMEOUT_MAX_MS - VR_DEBOUNCE_TIMER_TIMEOUT_MIN_MS) * (1.0 - appreciationIntensity);
+        // This timer forces a minimum tail duration for all sound clips
+        if (!vrDebounceTimer) {
+            selectAndPlaySound();
+            vrDebounceTimer = Script.setTimeout(function() {
+                vrDebounceTimer = false;
+            }, vrDebounceTimeout);
+        }
+    }
+
+    // Gets both hands' linear velocity.
+    var lastLeftHandPosition = false;
+    var lastRightHandPosition = false;
+    function getHandsLinearVelocity() {
+        var returnObject = {
```

Comment:
it's pretty common to use var names similar to this when you KNOW that that is the var that'll be returned. you can usually determine what the value is based on the function name or context. buuuuuuut...just because it's common doesn't mean it's best :D i'll change it, thanks for commenting